tests/e2e/utils.go

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+const (
+	kubectlNotFoundErrorMsg = "NotFound"
+)
+
+// stringSlicesUnion returns the union of the slices from argument.
+func stringSlicesUnion(sliceA, sliceB []string) []string {
+	if len(sliceA) == 0 || len(sliceB) == 0 {
+		return nil
+	}
+
+	sort.Slice(sliceA, func(i int, j int) bool {
+		return sliceA[i] < sliceA[j]
+	})
+	sort.Slice(sliceB, func(i int, j int) bool {
+		return sliceB[i] < sliceB[j]
+	})
+
+	var union []string
+	for i := range sliceB {
+		for j := range sliceA {
+			if sliceB[i] < sliceA[j] {
+				break
+			} else if sliceB[i] == sliceA[j] {
+				union = append(union, sliceB[i])
+			}
+		}
+	}
+	return union
+}
```

Comment:
**Required**: This **MUST** be fixed because if we are looking for the union (merge and reduce) of the two slices the current solution fails on the following inputs:
["1","22","333"], ["4444","55555","666666"] => []
when it should result in ["1", "22", "333", "4444", "55555", "666666"].

What your function does instead is to filter the sliceB to only those elements which can be found in sliceA which is the intersect not the union.
If that's what we want to do then the function seems good, but should be renamed accordingly.

Also is this used anywhere? I don't recall but don't have time to check back just now.

Reply:
It is used multiple places

Reply:
fixed: [cc6b1fc](https://github.com/banzaicloud/koperator/pull/987/commits/cc6b1fc6950f8408a00aa1ebba0dc2c941315531)